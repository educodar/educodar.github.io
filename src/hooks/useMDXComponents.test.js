import renderer from "react-test-renderer"
import useMDX from "./useMDXComponents"

describe("Use MDX Components", () => {
  it("returns a object", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toBeInstanceOf(Object)
  })

  it("returns a object with 'a' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("a")
  })

  it("returns a object with 'blockquote' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("blockquote")
  })

  it("returns a object with 'h1' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("h1")
  })

  it("render correctly the 'h1' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.h1({ test: "h1 component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'h2' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("h2")
  })

  it("render correctly the 'h2' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.h2({ test: "h2 component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'h3' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("h3")
  })

  it("render correctly the 'h3' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.h3({ test: "h3 component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'h4' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("h4")
  })

  it("render correctly the 'h4' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.h4({ test: "h4 component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'h5' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("h5")
  })

  it("render correctly the 'h5' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.h5({ test: "h5 component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'h6' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("h6")
  })

  it("render correctly the 'h6' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.h6({ test: "h6 component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'hr' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("hr")
  })

  it("returns a object with 'img' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("img")
  })

  it("returns a object with 'li' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("li")
  })

  it("render correctly the 'li' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.li({ test: "li component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'p' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("p")
  })

  it("render correctly the 'p' component returned with custom props", () => {
    const componentsObject = useMDX()

    const tree = renderer
      .create(componentsObject.p({ test: "p component" }))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("returns a object with 'table' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("table")
  })

  it("returns a object with 'td' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("td")
  })

  it("returns a object with 'tr' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("tr")
  })

  it("returns a object with 'ul' property", () => {
    const componentsObject = useMDX()

    expect(componentsObject).toHaveProperty("ul")
  })
})
