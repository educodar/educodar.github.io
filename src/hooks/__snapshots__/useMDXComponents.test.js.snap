// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use MDX Components render correctly the 'h1' component returned with custom props 1`] = `
<h1
  className="MuiTypography-root MuiTypography-h1"
  test="h1 component"
/>
`;

exports[`Use MDX Components render correctly the 'h2' component returned with custom props 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h2"
  test="h2 component"
/>
`;

exports[`Use MDX Components render correctly the 'h3' component returned with custom props 1`] = `
<h3
  className="MuiTypography-root MuiTypography-h3"
  test="h3 component"
/>
`;

exports[`Use MDX Components render correctly the 'h4' component returned with custom props 1`] = `
<h4
  className="MuiTypography-root MuiTypography-h4"
  test="h4 component"
/>
`;

exports[`Use MDX Components render correctly the 'h5' component returned with custom props 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5"
  test="h5 component"
/>
`;

exports[`Use MDX Components render correctly the 'h6' component returned with custom props 1`] = `
<h6
  className="MuiTypography-root MuiTypography-h6"
  test="h6 component"
/>
`;

exports[`Use MDX Components render correctly the 'li' component returned with custom props 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemText-root"
    test="li component"
  />
</li>
`;

exports[`Use MDX Components render correctly the 'p' component returned with custom props 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
  test="p component"
/>
`;
